### PR TITLE
add default implementation for Usertype

### PR DESCRIPTION
**Pull request by Invincible2211**

### CODE REVIEW

File: src/main/java/de/yggdrasil/core/dal/DAL.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import de.yggdrasil.core.dal.response.Response;
 import de.yggdrasil.core.dal.util.LogoLogger;
 import de.yggdrasil.core.dal.util.profile.DALProfile;
+import de.yggdrasil.core.dal.util.profile.ProfileProduction;
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/DAL.java:11]: Unnecessary import?

File: src/main/java/de/yggdrasil/core/dal/datasource/models/usertype/DefaultUserType.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * A generic interface that extends Hibernate UserType to handle JSON serialization and deserialization
+ * of custom data types. This interface ensures data can be stored as JSON strings in a database using
+ * VARCHAR SQL type and provides methods for seamless data conversion and operation integrity.
+ */
+@HibernateUserType
+public interface DefaultUserType<T> extends UserType<T> {
+
+    Logger logger = LoggerFactory.getLogger(DefaultUserType.class);
+    Gson gson = new Gson();
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/datasource/models/usertype/DefaultUserType.java:28]: You might want to make these variables static, as far as I know reusing them shouldn't lead to any problems

File: src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package de.yggdrasil.core.dal.logging.error;
+
+public interface ErrorLoggingUserType {
+
+    String TYPE_PARAMETER_CLASS_DETERMINATION_FAILED = "Failed to determine the class type parameter.";
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java:5]: Maybe use "generic type parameter"

File: src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    String JSON_SERIALIZATION_FAILED = "JSON serialization failed.";
+    String JSON_DESERIALIZATION_FAILED = "JSON deserialization failed.";
+    String OBJECT_COPY_FAILED = "Failed to create a deep copy of the object.";
+    String OBJECT_SERIALIZATION_FAILED = "Object serialization for caching/storing failed.";
+    String OBJECT_DESERIALIZATION_FAILED = "Object deserialization from cached/stored data failed.";
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java:10]: If there is no other way that a object serialization is performed then I don't think that you need the extra context for what it was

**Review by Morazzer — CHANGES_REQUESTED** [review]

File: src/main/java/de/yggdrasil/core/dal/datasource/models/usertype/DefaultUserType.java (hunk 2 above)
Comment (Invincible2211) [reply src/main/java/de/yggdrasil/core/dal/datasource/models/usertype/DefaultUserType.java:28]: They should already be static due to the interface context.

File: src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java (hunk 4 above)
Comment (Invincible2211) [reply src/main/java/de/yggdrasil/core/dal/logging/error/ErrorLoggingUserType.java:10]: It can occur both there and in the nullSafeGet/Set methods.

**Review by Morazzer — APPROVED** [review]

**Pull request merged**